### ISSUE TITLE
ckEditor 5

### ISSUE THREAD

**Issue by JosephMDavis** [i7MEDIA/mojoportal#230]

**Comment by ElijahFowler** [comment]
Thank you for your work on getting this moving.

Looking over the branch I spot a handful of things that need to addressed.

1. Pressing "Enter" (with any key combo) submits the form. We cannot insert a new paragraph at the moment.
2. We need to be able to use mojo's file picker and file manager to insert links and images into the content.
3. We need to be able to use custom configuration files similar to CKEditor 4 to allow configuration without pasting a bunch of JSON in the web config settings (long term plan is to make the web config as lean as possible).
4. We need to support mojo's Content Templates and CSS Style Templates features (see CkEditorStyles.ashx, CkEditorTemplates.ashx).

Thanks,
Elijah

**Comment by jedelfraisse** [comment]
I have the start of the version replacement in the branch.  I made a new editor (CKEditor5) as there was big change in the script to run it.  Do we want to put options in the Content Editor Admin screen to customize the CKEditor, such as license key for their paid things?

**Comment by jedelfraisse** [comment]
Thank you for testing.  There was a major release, but of things changed.  It at least has it started.   I will look into using the mojo file picker, Templates, etc.  For some of the settings, we could put them into the main settings for each site could have it own.

**Comment by jedelfraisse** [comment]
Just had another thought.  If developers want to use the implementation of ckEditor in their own add-ons, they should be able to specify what buttons are visible when calling the object.

**Comment by JosephMDavis** [comment]
> Just had another thought. If developers want to use the implementation of ckEditor in their own add-ons, they should be able to specify what buttons are visible when calling the object.

Agreed. Would be good to allow for this from the SkinConfig (skinname/config/config.js) as well for designers.
